scripts/run_phpunit_tests.sh

```diff
@@ -4,14 +4,82 @@ set -eux
 docker_name=$1
 dummy_container="dummy_$(date +%s)"
 
-# Create a dummy container which will hold a volume with source
-docker create -v /opt --name $dummy_container $docker_name /bin/true
+VOLUME=/opt
+WORK_DIR=${VOLUME}/project
+
+UNITTEST_REPORT_DIR=.phpunit/phpunit
+INTGTEST_REPORT_DIR=.phpunit/phpunit.integration
 
+mkdir -p ${PWD}/${UNITTEST_REPORT_DIR}
+mkdir -p ${PWD}/${INTGTEST_REPORT_DIR}
+
+# Create a dummy container which will hold a volume with source
+docker create -v ${VOLUME} --name $dummy_container $docker_name /bin/true
 # Copy source to dummy container
-docker cp $(pwd)  $dummy_container:/opt/project
+docker cp $(pwd) $dummy_container:${WORK_DIR}
+
 
 # Check syntax
-docker run -t -w /opt/project --volumes-from $dummy_container $docker_name /bin/bash -c 'a=$(find /opt/project -type f -name "*.php" !  -path "*/vendor/*" -print0 | xargs -0 -n 1 -P 8 php -l | grep -v "No syntax errors" | wc -l) && exit $a'
+docker run --rm -t -w ${WORK_DIR} --volumes-from $dummy_container $docker_name \
+       /bin/bash -c 'a=$(find /opt/project -type f -name "*.php" !  -path "*/vendor/*" -print0 | xargs -0 -n 1 -P 8 php -l | grep -v "No syntax errors" | wc -l) && exit $a'
+
+
+# Run unit test
+if [[ "${docker_name}" =~ ^php:7.*$ ]]; then
+    docker run -t -w ${WORK_DIR} --volumes-from $dummy_container $docker_name \
+           /bin/bash -c "set -e; phpdbg -qrr vendor/bin/phpunit --log-junit ${UNITTEST_REPORT_DIR}/results.xml -d memory_limit=1024M --coverage-html ${UNITTEST_REPORT_DIR}/coverage-report"
+else
+    docker run -t -w ${WORK_DIR} --volumes-from $dummy_container $docker_name \
+           /bin/bash -c "set -e; vendor/bin/phpunit --log-junit ${UNITTEST_REPORT_DIR}/results.xml"
+fi
+docker cp $dummy_container:"${WORK_DIR}/${UNITTEST_REPORT_DIR}" ${PWD}/${UNITTEST_REPORT_DIR}
+
+
+# Replace for Integration test
+if [ "${docker_name}" == "vectorface/php5.4" ]; then
+    docker_name="php:5.4-apache"
+    for n in $(seq 1 10); do docker pull $docker_name && break; done
+elif [ "${docker_name}" == "vectorface/php5.3" ]; then
+    docker_name="php:5.3-apache"
+    # FIX: Unable to find image 'php:5.3-apache' locally and Error response from daemon: unexpected EOF
+    for n in $(seq 1 10); do docker pull $docker_name && break; done
+elif [ "${docker_name}" == "vectorface/hhvm" ]; then
+    # TODO: Not supported yet.
+    echo Not supported integration test for hhvm
+    exit 0
+fi
+
+# Run Apache for integration test
```

Comment:
I prefer to split integration test from `scripts/run_phpunit_tests.sh`.

Reply:
@Soichiro725 
In the case of a separate Integration Test, the creation of a dummy container and a code copy process must be performed again.
Initially they were isolated, but these did not help improve the speed of the test.
So I thought one script would be enough, what do you think?

Reply:
I feel `scripts/run_phpunit_tests.sh` is too complex.
WOVN.php test is not so slow now.
And you can run integration tests in parallel by defining another workflows of circleCI.